devito/ops/types.py

```diff
@@ -0,0 +1,15 @@
+from devito.tools import Signer
+
+import devito.types.basic as basic
+
+__all__ = ['Array']
+
+
+class Array(basic.Array, Signer):
+
+    from_OPS = True
+    is_Symbol = True
```

Comment:
the entire PR looks good to go to me except for this ``is_Symbol`` -- an Array definitely is not a symbol! Why do you need it ? Maybe we can find a solution by introducing a new subclass...

Reply:
reminder

Reply:
What I think you should do is:

* call it differently: SymbolicArray ? remind me again (sorry) what C symbols it is supposed to represent?
* Add is_Tensor = False
* unflip the change in ``devito/ir/iet/visitors.py``

Remind me why can you not just inherit from Symbol and change (override) what's required?